src/backend/distributed/sql/udfs/get_missing_time_partition_ranges/latest.sql

```diff
@@ -0,0 +1,264 @@
+CREATE OR REPLACE FUNCTION pg_catalog.get_missing_time_partition_ranges(
+    table_name regclass,
+    to_date timestamptz,
+    start_from timestamptz DEFAULT now(),
+    partition_interval INTERVAL DEFAULT NULL)
+returns table(
+    partition_name text,
+    range_from_value text,
+    range_to_value text)
+LANGUAGE plpgsql
+AS $$
+DECLARE
+    distinct_partition_interval_count int;
+    table_partition_interval INTERVAL;
+    is_multiple_days boolean;
+    table_partition_column_type_name text;
+    manual_partition_from_value_text text;
+    manual_partition_to_value_text text;
+    current_range_from_value timestamptz := NULL;
+    current_range_to_value timestamptz := NULL;
+    current_range_from_value_text text;
+    current_range_to_value_text text;
+    datetime_string_format text;
+    max_table_name_length int;
+    table_name_text text;
```

Comment:
very minor / optional: given the wall of declarations, some comments on their purpose (possibly in groups) could be helpful.